.github/workflows/ci.yml

```diff
@@ -110,3 +112,38 @@ jobs:
         timeout_minutes: 15
         max_attempts: 2
         command: ./tools/test_monitor/run-tests.sh
+  flow-legacy-client-compatibility-test:
+    name: Client Compatibility Tests With Observer
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version:
+          - 1.16
+          - 1.17
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ matrix.go-version}}
+    - name: Checkout repo
+      uses: actions/checkout@v2
+    - name: Build relic
+      run: make crypto_setup_gopath
+    - name: Docker build
+      run: make install-tools
+    - name: Install Flow Client
+      run: sh -ci "$(curl -fsSL https://storage.googleapis.com/flow-cli/install.sh)" && mv ~/.local/bin/flow /tmp
+    - name: Set up Localnet
+      run: bash -c 'cd integration/localnet/ && make -e OBSERVER=1 init && make start'
+    - name: Ensure Observer is started
+      run: docker ps -q -f name=localnet_observer_1_1
+    - name: Get Version ensuring the official client build can connect
+      run: /tmp/flow -f integration/localnet/flow-localnet.json -n observer version
+    - name: Get Status ensuring the official client build can connect
+      run: /tmp/flow -f integration/localnet/flow-localnet.json -n observer status
+    - name: Wait for finalized blocks and check them with Observer
+      # || true = non-breaking due to rpc change. We do access and observer together to fetch the same block
+      run: /tmp/flow -f integration/localnet/flow-localnet.json -n access blocks get latest && (/tmp/flow -f integration/localnet/flow-localnet.json -n observer blocks get latest || true)
```

Comment:
Can you explain the `|| true` more?

Reply:
It is from the line above "non-breaking due to rpc change. We do access and observer together to fetch the same block"

The current version in master does not reply properly to the rpc request. We would like to create a test that does not break in case of such a branch, but it still provides meaningful debugging information. This is temporary. You can see the final test here after the gRPC proxy is merged in the third pull request:
https://github.com/GetElastech/flow-go/blob/cbdf0ca3e5e4e10529dff7a82ebab5c757cc468d/.github/workflows/ci.yml#L146

While the `sleep 5` was objected as a separate item, the containers start up, and we have seen one flaky case that supports adding it. We have unit tests that we add as a separate branch that does a similar check.